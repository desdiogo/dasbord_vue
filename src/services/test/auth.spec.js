import mockAxios from "axios";
import AuthService from "../auth";

jest.mock("axios");

describe("auth services", () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  it("Login com sucesso", async () => {
    const token = "12346789Oi";
    mockAxios.post.mockImplementationOnce(() => {
      return Promise.resolve({ data: { token } });
    });

    const response = await AuthService(mockAxios).login({
      email: "igor@igor.me",
      password: "1234",
    });
    expect(response.data).toHaveProperty("token");
    expect(response).toMatchSnapshot();
  });

  it('Retornando dados do login', async () => {
    const user = {
      name: 'Igor',
      password: '1234',
      email: 'igor@igor.me'
    }
    mockAxios.post.mockImplementationOnce(() => {
      return Promise.resolve({ data: user })
    })

    const response = await AuthService(mockAxios).register(user)
    expect(response.data).toHaveProperty('name')
    expect(response.data).toHaveProperty('password')
    expect(response.data).toHaveProperty('email')
    expect(response).toMatchSnapshot()
  })

  it('Erro no login', async () => {
    const errors = { status: 404, statusText: 'Not Found' }
    mockAxios.post.mockImplementationOnce(() => {
      return Promise.resolve({ request: errors })
    })

    const response = await AuthService(mockAxios).login({ email: 'igor@igor.me', password: '123' })
    expect(response.errors).toHaveProperty('status')
    expect(response.errors).toHaveProperty('statusText')
  })
});
