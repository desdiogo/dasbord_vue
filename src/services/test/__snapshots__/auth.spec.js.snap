// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth services Login com sucesso 1`] = `
Object {
  "data": Object {
    "token": "12346789Oi",
  },
  "errors": null,
}
`;

exports[`auth services Retornando dados do login 1`] = `
Object {
  "data": Object {
    "email": "igor@igor.me",
    "name": "Igor",
    "password": "1234",
  },
  "errors": null,
}
`;
